fix(neogit): improve Neogit colors

mizlan opened the pull request:
## before

<img width="1481" alt="Screen Shot 2022-08-06 at 5 52 43 PM" src="https://user-images.githubusercontent.com/44309097/183270515-5a90e056-9f1b-46d0-9905-4c8f14a1bb7e.png">

## after

<img width="1232" alt="Screen Shot 2022-08-06 at 5 40 42 PM" src="https://user-images.githubusercontent.com/44309097/183270525-7534a092-391b-4755-b403-a449d00f09ca.png">

mizlan commented:
As an active Neogit user, I find the existing colors intolerable so I created this new set of highlights that takes advantage of `ucolors.darken` a lot!

nullchilly commented:
Thank! I am still using vim-fugitive
I guess its time to try neogit out :)

nullchilly commented:
@mizlan Don't you think it's still a bit hard to see in NeogitCommitView?
![image](https://user-images.githubusercontent.com/56817415/183271154-f344b2b9-848a-42c5-b34a-593aa71d2024.png)

nullchilly commented on lua/catppuccin/groups/integrations/neogit.lua:
```diff
@@ -1,14 +1,30 @@
+		NeogitHunkHeader = {
+			bg = ucolors.darken(cp.blue, 0.095, cp.base),
+			fg = ucolors.darken(cp.blue, 0.5, cp.base),
+		},
```
Can you add NeogitCommitViewHeader?

mizlan replied:
got it.

mizlan commented:
Also see this: https://github.com/TimUntersberger/neogit/issues/330

Will work on this, will also probably PR Neogit to get customization on the other hunks (I want to dim unfocused hunk's context lines)

mizlan commented:
<img width="618" alt="Screen Shot 2022-08-06 at 6 38 00 PM" src="https://user-images.githubusercontent.com/44309097/183271366-c0ac3fef-7b19-47b5-87a2-b85c9d797fcb.png">

Foreground also turns weird blue color in Commit view mode, will try to fix as well

mizlan commented:
This is the best I can do for now, since I dont know why it's blue. Fixed formatting, and added some more hlgroups. Take a look:
<img width="753" alt="Screen Shot 2022-08-06 at 8 48 05 PM" src="https://user-images.githubusercontent.com/44309097/183274348-b8201c2f-2497-472b-b8c0-3fcc5f8e415e.png">
<img width="742" alt="Screen Shot 2022-08-06 at 8 48 29 PM" src="https://user-images.githubusercontent.com/44309097/183274360-a4ba008d-0d9d-443a-bfc9-ab9330f28cf8.png">

nullchilly commented:
@mizlan It's too dim for me though?
![image](https://user-images.githubusercontent.com/56817415/183274481-26092f36-f2e6-48cd-a64b-43d4939a85c3.png)

mizlan commented:
Yeah, thats a bit of a problem. However, the goal is to fix that in Neogit itself so the hunk under cursor uses the other set of highlights. These, then, would be the "inactive" hunks. I think for now this is an alright approach.

nullchilly commented:
@mizlan So we are missing the ActiveHunk

Why not just decrease the dim in the meantime and open another pr later when InactiveHunk is fixed?

mizlan commented:
Good question. Maybe my workflow is different than others but I spend 99% of my time in the status buffer, where the dimmed text matters and is significantly helpful. I rarely view existing commits or browse diffs in other forms in Neogit. I cannot say the same for others' workflows but I'll be happy to change it if you prefer. I could leave those commented out with a note to re-enable them after Neogit is updated.

nullchilly commented:
Yeah that sounds good

Remember we try to not be opinionated as much as possible

Personally reference is done via custom_highlight :)

mizlan commented:
Oh. I've got an idea. I'll use a more vibrant bg for "active" hunks right now. Will be easily visibly distinguishable while allowing both active and inactive to be readable.

mizlan commented:
<img width="989" alt="Screen Shot 2022-08-06 at 9 26 39 PM" src="https://user-images.githubusercontent.com/44309097/183275269-1c22c3ec-08a7-42ba-8a36-fa389c0cef6a.png">
Quite fond of this setup: both are legible but one is grabs my attention more.

nullchilly commented:
@mizlan Seems good!

We can merge this now if you want?

mizlan commented:
Sure, go ahead! Can't find anything else for now.
